libs/blocks/commerce/commerce.js

```diff
@@ -129,18 +130,21 @@ export const decorateOfferDetails = async (el, of, searchParams) => {
 export const handleSearch = async (event, el) => {
   let searchParams = {};
   const displaySearchError = () => {
-    const notValidUrl = document.createElement('h4');
-    notValidUrl.classList.add('not-valid-url');
-    notValidUrl.textContent = 'Not a valid offer link';
+    const notValidUrl = createTag('h4', { class: 'not-valid-url' }, 'Not a valid offer link')
     el.append(notValidUrl);
   };
   el.textContent = '';
   const search = event.target.value;
-  searchParams = new URL(search).searchParams;
+  try {
```

Comment:
```suggestion
  const osi = new URLSearchParams(search).get('osi');
  if (!osi) {
    displaySearchError();
    return;
  }
```

Reply:
love this suggestion! - done

Reply:
@narcis-radu I just discovered that URLSearchParams don't properly if you pass a full URL in as the param. The proposal didn't therefore quite work, but good to know for the future.  

- https://stackoverflow.com/questions/55103757/urlsearchparams-returning-null-for-the-first-query-string
- https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#no_url_parsing